lib/rules/lint-table-groups.js

```diff
@@ -19,18 +19,31 @@ function hasDisallowedChildren(tableNode) {
     [],
     validBlocks.map(block => [].concat(block.program.body, block.inverse ? block.inverse.body : []))
   );
-  return []
+  return ![]
     .concat(
       childrenFromValidBlocks,
       tableNode.children.filter(item => validBlocks.indexOf(item) === -1)
     )
-    .some(
-      child =>
-        !(
-          (child.type === 'ElementNode' && ALLOWED_TABLE_CHILDREN.indexOf(child.tag) > -1) ||
-          (child.type === 'TextNode' && !/\S/.test(child.chars))
-        )
-    );
+    .every(isAllowedTableChild);
+}
+
+function isAllowedTableChild(node) {
+  let tagNamePair;
+
+  switch (node.type) {
+    case 'MustacheStatement':
+      tagNamePair = node.hash.pairs.filter(pair => pair.key === 'tagName')[0];
+      if (tagNamePair) {
+        return ALLOWED_TABLE_CHILDREN.includes(tagNamePair.value.value);
+      }
+      break;
+    case 'ElementNode':
+      return ALLOWED_TABLE_CHILDREN.includes(node.tag);
```

Comment:
This case needs some of the same treatment as above (for angle bracket invocation), can you also add a test with:

```hbs
<table>
  <SomeThing @tagName="thead" />
</table>
```

Reply:
updated